controller/volume_controller.go

```diff
@@ -3054,6 +3066,45 @@ func (vc *VolumeController) processMigration(v *longhorn.Volume, es map[string]*
 	return nil
 }
 
+func (vc *VolumeController) IsReplicaUnavailable(r *longhorn.Replica) (bool, error) {
+	log := vc.logger.WithFields(logrus.Fields{
+		"replica":       r.Name,
+		"replicaNodeID": r.Spec.NodeID,
+	})
+
+	if r.Spec.FailedAt != "" {
+		return true, nil
+	}
+	if r.Spec.NodeID == "" || r.Spec.DiskID == "" {
+		return true, nil
+	}
+
+	isDownOrDeleted, err := vc.ds.IsNodeDownOrDeleted(r.Spec.NodeID)
+	if err != nil {
+		log.WithError(err).Errorf("Unable to check if node %v is still running for failed replica", r.Spec.NodeID)
+		return true, err
+	}
+	if isDownOrDeleted {
+		return true, nil
+	}
```

Comment:
NIT: can combine these, since err is nil when `isDownOrDeleted`

Reply:
Although the returned boolean is true in both cases, the log will be printed only when the err is not nil. I prefer not to retain the log rather than combining these like:
```
if isDownOrDeleted, err := vc.ds.IsNodeDownOrDeleted(r.Spec.NodeID); err != nil || isDownOrDeleted{
    return true, err
}
```